src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/LocalHelpers.cs

```diff
@@ -104,6 +104,47 @@ public static string GetRepoPathFromFolder(string sourceFolder, string commit, I
             return null;
         }
 
+        /// <summary>
+        /// Since LibGit2Sharp doesn't support neither sparse checkout not shallow clone
+        /// we implement the flow ourselves.
+        /// </summary>
+        /// <param name="repoUri">The repo to clone Uri</param>
+        /// <param name="branch">The branch to checkout</param>
+        /// <param name="workingDirectory">The working directory</param>
+        /// <param name="_logger">The logger</param>
+        /// <param name="remote">The name of the remote</param>
+        /// <param name="user">User name</param>
+        /// <param name="pat">User's personal access token</param>
+        /// <param name="repoFolderName">The name of the folder where the repo is located</param>
+        /// <returns>The full path of the cloned repo</returns>
+        public static string SparseAndShallowCheckout(
+            string repoUri,
+            string branch,
+            string workingDirectory,
+            ILogger _logger,
+            string remote,
+            string user,
+            string pat,
+            string repoFolderName = "clonedRepo")
+        {
+            Directory.CreateDirectory(workingDirectory);
+
+            ExecuteGitShallowSparseCommand("git", $"init {repoFolderName}", _logger, workingDirectory);
+
+            workingDirectory = Path.Combine(workingDirectory, repoFolderName);
+            repoUri = repoUri.Replace("https://", $"https://{user}:{pat}@");
+
+            ExecuteGitShallowSparseCommand("git", $"remote add {remote} {repoUri}", _logger, workingDirectory);
+            ExecuteGitShallowSparseCommand("git", "config core.sparsecheckout true", _logger, workingDirectory);
+            ExecuteGitShallowSparseCommand("cmd", "/c echo eng/ >> .git/info/sparse-checkout", _logger, workingDirectory);
```

Comment:
Using cmd for all of these commands would mean that we wouldn't be able to run the service locally in linux, is that something we should take into consideration?

Reply:
Done